Add GitHub Actions config file

abetomo opened the pull request:
You can also test on Windows.

# Pull Request

<!--
The text in these markdown comments is instructions that will not appear in the displayed pull request,
and can be deleted.

Please submit pull requests against the develop branch.

Follow the existing code style. Check the tests succeed, including lint.
  npm run test
  npm run lint

Don't update the CHANGELOG or command version number. That gets done by maintainers when preparing the release.

Commander currently has zero production dependencies. That isn't a hard requirement, but is a simple story. Requests which 
add a dependency are much less likely to be accepted, and we are likely to ask if there alternative approaches to avoid the dependency.
-->

## Problem

We were not using Windows CI.

## Solution

Use GitHub Actions to test on Windows.

## ChangeLog

<!--
Optional. Suggest a line for adding to the CHANGELOG to summarise your change.
-->

shadowspawn commented:
Nice! I will research a bit about Github Actions before reviewing. 

(There will likely be some minor code changes to get current tests running and passing on Windows: #916 )

shadowspawn commented:
3.0.1 is out, so keen to give this a go! (Did you sign up for the beta?)
I expect the Windows build will break, but that is ok.
Not sure if this will be active from develop branch, but will try that first.

Thanks @abetomo. I was thinking of the Azure PR we got a while ago but I like this better.

abetomo commented:
Renamed the configuration file, and fixed the problem with the settings.

This was the result for my repository.
https://github.com/abetomo/commander.js/runs/207808904

This repository is also valid, so it will work if merged.

abetomo commented:
https://github.com/tj/commander.js/runs/207817593
It seems that this repository worked when merged with develop.

shadowspawn commented:
Nice! How do you see the runs? (When I go to Actions I just see templates, but I can follow your link and see that run.)

abetomo commented:
Probably, you can see it from the 'Actions' tab once it has been merged with master.

shadowspawn commented:
I have signed up for the Actions beta, I might be able to see more when accepted.

It does not look like the test are running on Windows, but I was not sure why on first look.

(Woop woop! Thanks.)

abetomo commented:
It's hard to see in the capture, but it seems to work on Windows.

<img width="1007" alt="スクリーンショット 2019-08-30 18 53 34" src="https://user-images.githubusercontent.com/846454/64011904-8b3af180-cb57-11e9-8d88-5cd9c0e10e74.png">

shadowspawn commented:
I got added to the beta program, and can see the past workflow runs now. I am still suspicious whether the Windows test are running but clearly they are meant to. Looking forward to merging with the Jest branch. Exciting!
